src/runtime_src/tools/scripts/build_xrt_w_docker.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2019-2022 Xilinx, Inc. All rights reserved.
+#
+
+# Must be Ubuntu 18.04 or Ubuntu 20.04
+BASE_IMAGE=$1:-ubuntu:18.04
+
+SCRIPT_DIR=$(dirname $(readlink -f $0))
+
+REPO_DIR=$(readlink -f $SCRIPT_DIR/../../../../)
+
+XRTDEPS_DIR=$SCRIPT_DIR
+
+BUILD_DIR=$REPO_DIR/build
+
+echo "Creating xrt-build-ubuntu Docker Image with xrtdeps.sh"
+
+CONTAINER_ID=$(docker run -d -e DEBIAN_FRONTEND=noninteractive -v $REPO_DIR:$REPO_DIR -w $XRTDEPS_DIR ubuntu:18.04 /bin/bash -c " apt-get update && ./xrtdeps.sh")
+
+docker wait $CONTAINER_ID
+
+docker commit $CONTAINER_ID xrt-build-ubuntu
+
+echo "Building XRT inside of Docker container"
+echo "Outputs will be in: $BUILD_DIR"
```

Comment:
You are at most creating a directory under BUILD_DIR, right?
Will that directory conflict with my native host folders Debug, Release, etc?  IN other words should you add a subfolder 'docker-<image>' maybe?

Reply:
@stsoe no build directory is created.

The native host folder is mounted into the container `-v $REPO_DIR:$REPO_DIR`. This means that the absolute path of XRT will be mounted... for instance... it may be `/home/bryanloz/XRT`. This path will be visible both outside, and inside the container. `$BUILD_DIR` is just used to communicate the working directory.

The build outputs will get generated as usual in `/home/bryanloz/XRT/build/Debug` and `/home/bryanloz/XRT/build/Release`

In this approach the docker container will work seamlessly with the native host filesystem.

Reply:
@bryanloz-xilinx My point was that I may want to build XRT on my native host as well as docker.  I don't want docker to re-use my native builds Release and Debug directories, so move the docker build dirs one level down under a docker-<image> subdir is what I suggested.

Reply:
Bryan, please also rename with the script :-) what's the point of saving 3 characters?

Reply:
Script is renamed.

Docker is just calling XRT's `XRT/build/build.sh`, the output will go wherever that script wants to put it.

Is there a way to tell `XRT/build/build.sh` to put the build products somewhere else? If so, I can accommodate.

Reply:
Is it safe to play with these variables?
https://github.com/Xilinx/XRT/blob/master/build/build.sh#L188-L190

Reply:
Ahh yeah, got you about just using build.sh default locations.
Yes, to accommodate DevOps custom build dirs, env variables can be set prior to invoking build.sh.
DEBUG_DIR,  REL_DIR,  EDGE_DIR, so yes, use these.